### ISSUE TITLE
Double messaging members

### ISSUE THREAD

**Issue by rapala61** [DoSomething/gambit-conversations#331]
_BUG_
-------------------
#### Current Behavior
- Under some circumstances. Members get 2 messages instead of 1. Twilio is reporting 2 separate "delivered" events for the same member. The second delivered message is not stored in G-Conversations.

#### Desired Behavior
- Members should only get 1 message under any circumstances. 

#### Steps to Replicate
- WIP

#### Why This Matters
- It's breaking Blink.
- It's sending members 2 messages so it's annoying and double the expected cost.

#### Relevant Screenshots + Links
- This bug was reported in Blink first. Most of the sleuthing results are stored there: https://github.com/DoSomething/blink/issues/203

**Comment by rapala61** [comment]
The Twilio team was contacted to see if the behavior was potentially due to something on their end.

**Comment by rapala61** [comment]
[Saved log search with the `parseMessageIdFromBody()` error](https://f1dc9460e3fc9995e3105d273237338a.us-east-1.aws.found.io:9243/app/kibana#/discover/776c9600-65ae-11e8-8dbf-cbfcda283cee?_g=(refreshInterval%3A(display%3AOff%2Cpause%3A!f%2Cvalue%3A0)%2Ctime%3A(from%3A'2018-03-01T05%3A00%3A00.000Z'%2Cmode%3Aabsolute%2Cto%3A'2018-05-06T05%3A27%3A10.996Z'))). April 30th is the first time we see this error.

**Comment by ngjo** [comment]
Card: https://www.pivotaltracker.com/story/show/161604351

**Comment by rapala61** [comment]
Fixed in https://github.com/DoSomething/gambit-conversations/pull/447

**Comment by rapala61** [comment]
Not fixed yet! Turns out this bug seems to be caused by timeout requests that Blink retries but that actually went through:

From the Slack post detailing this problem based on observed high number of retries from last full list broadcast 12/05/2018:

**Bad news**:

The duplicate messages are back, the PR I merged a few days ago doesn't seem to have done the trick under a full list broadcast.

**Good news**:

Investigated through the last message for users that display this behavior. It seems like ~the~ there is a pattern where the last outbound message was retried once before going through. That is one clue I did not uncover before and can help pin point the root cause of this.

## My updated theory based on the above finding

### TL;DR
A retried message caused by timeouts causes:
- a double message to be sent to some users
- a race condition between Twilio delivery receipts. The receipt that can't find it's message errors out, causing increased 500 errors.

### Summary

The request from the broadcast worker to G-Conv gets timed out by Heroku at the 30 seconds mark. G-Conv actually finishes processing the request since it already queued it internally, which sends the message to the user. Blink thinks it failed so it retries it.

The retry is received by G-Conv and because the message w/ that `request_id` already exists (which is the way it finds the messages on retries) it loads it, updates the retry count metadata and re-sends the message to the user... BUT, this time, the delivery receipt of the LAST processed request is the one that will win and update the delivery metadata of the message.

The delivery metadata belonging to the first outbound will not find a MessageSid and will error out, causing the 500 error in blink to be retried.
